ouroboros-network/src/Ouroboros/Network/Diffusion/MinimalP2P.hs

```diff
@@ -0,0 +1,893 @@
+              ( catMaybes
+                [ daIPv4Address
+                , daIPv6Address
+                ]
+              )
+              f
+
+          -- run server
+          withServer sockets connectionManager inboundInfoChannel =
+            Server.with
+              Server.Arguments {
+                  Server.sockets               = sockets,
+                  Server.snocket               = diNtnSnocket,
+                  Server.tracer                = dtServerTracer,
+                  Server.trTracer              = dtInboundGovernorTransitionTracer,
+                  Server.debugInboundGovernor  = nullTracer,
+                  Server.inboundGovernorTracer = dtInboundGovernorTracer,
+                  Server.connectionLimits      = daAcceptedConnectionsLimit,
+                  Server.connectionManager     = connectionManager,
+                  Server.connectionDataFlow    = diNtnDataFlow,
+                  Server.inboundIdleTimeout    = Just daProtocolIdleTimeout,
+                  Server.inboundInfoChannel    = inboundInfoChannel
+                }
+
+      --
+      -- Part (b): capturing the major control-flow of runM:
+      --
+      case diffusionMode of
+
+        -- InitiatorOnly mode, run peer selection only:
+        InitiatorOnlyDiffusionMode ->
+          withConnectionManagerInitiatorOnlyMode $ \connectionManager-> do
+          debugStateVar <- newTVarIO $ emptyPeerSelectionState fuzzRng (CPST.empty caeConsensusMode caeMinBigLedgerPeersForTrustedState) CPRP.empty
+          diInstallSigUSR1Handler connectionManager debugStateVar daPeerMetrics
+          withPeerStateActions' connectionManager $ \peerStateActions->
+            withPeerSelectionActions'
+              (return Map.empty)
+              PeerSelectionActionsDiffusionMode { psPeerStateActions = peerStateActions } $
+              \(ledgerPeersThread, localRootPeersProvider) peerSelectionActions->
+                Async.withAsync
+                  (peerSelectionGovernor'
+                    dtDebugPeerSelectionInitiatorTracer
+                    debugStateVar
+                    peerSelectionActions) $ \governorThread ->
+                    Async.withAsync
+                      peerChurnGovernor' $ \churnGovernorThread ->
+                      -- wait for any thread to fail:
+                      snd <$> Async.waitAny
+                                [ledgerPeersThread, localRootPeersProvider, governorThread, churnGovernorThread]
+
+        -- InitiatorAndResponder mode, run peer selection and the server:
+        InitiatorAndResponderDiffusionMode -> do
+          inboundInfoChannel  <- newInformationChannel
+          withConnectionManagerInitiatorAndResponderMode
+            inboundInfoChannel $ \connectionManager ->
+              withSockets' $ \sockets addresses -> do
+                withServer sockets connectionManager inboundInfoChannel $ \inboundGovernorThread readInboundState -> do
```

Comment:
I'd keep the `-- node-to-node server` comment.

ref: a382ed87d2998ff90dd825ddbc6ac2ccbdac8da0